sql/src/test/java/io/crate/testing/RowSender.java

```diff
@@ -100,4 +105,38 @@ public int numPauses() {
     public int numResumes() {
         return numResumes;
     }
+
+    /**
+     * Generates N rows where each row will just have 1 integer column, the current range iteration value.
+     * N is defined by the given <p>start</p> and <p>end</p> arguments.
+     *
+     * @param start         range start for generating rows (inclusive)
+     * @param end           range end for generating rows (exclusive)
+     * @param reverse       if true, list of rows will be reversed before emitting
+     * @param step          range step
+     * @param rowReceiver   rows will be emitted on that RowReceiver
+     * @return              the last emitted integer value
+     */
+    public static int generateRowsInRangeAndEmit(int start, int end, boolean reverse, int step, RowReceiver rowReceiver) {
+        List<Row> rows = new ArrayList<>(end - start);
+        for (int i = start; i < end; i += step) {
+            rows.add(new RowN(new Object[]{i}));
```

Comment:
I think it would be cool to use a shared row (would require the rows to the built reverse as-well..)  but it would make sure that everything that is tested by this function handles shared rows correctly